torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py

```diff
@@ -88,12 +88,49 @@ def aot_dispatch_base_graph(
         fw_only=flat_fn,
     )
 
+    if aot_config.is_export:
+        assigned_buffers = {}
+
+        def _map_assigned_buffer_to_proxy(_mod, name, buffer):
+            proxy_mode = None
+            for mode in torch.utils._python_dispatch._get_current_dispatch_mode_stack():
```

Comment:
nit: you can do this:
```
torch._C._get_dispatch_mode(torch._C._TorchDispatchModeKey.PROXY)
```

Reply:
Thanks. 

As for asserting that the proxy mode always exists: unfortunately we don't have that guarantee right now. Saw test failures involving closures, cond, submodules, maybe more...

Reply:
More details here: https://github.com/pytorch/pytorch/pull/122337#discussion_r1560182916